src/Mvc/Mvc.TagHelpers/src/ComponentTagHelper.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Mvc.TagHelpers
+{
+    /// <summary>
+    /// A <see cref="TagHelper"/> that renders a Razor component.
+    /// </summary>
+    [HtmlTargetElement("component", Attributes = ComponentTypeName, TagStructure = TagStructure.WithoutEndTag)]
+    public class ComponentTagHelper : TagHelper
+    {
+        private const string ComponentParameterName = "params";
+        private const string ComponentParameterPrefix = "param-";
+        private const string ComponentTypeName = "type";
+        private const string RenderModeName = "render-mode";
+        private IDictionary<string, object> _parameters;
+
+        /// <summary>
+        /// Gets or sets the <see cref="Rendering.ViewContext"/> for the current request.
+        /// </summary>
+        [HtmlAttributeNotBound]
+        [ViewContext]
+        public ViewContext ViewContext { get; set; }
+
+        /// <summary>
+        /// Gets or sets values for component parameters.
+        /// </summary>
+        [HtmlAttributeName(ComponentParameterName, DictionaryAttributePrefix = ComponentParameterPrefix)]
+        public IDictionary<string, object> Parameters
+        {
+            get
+            {
+                _parameters ??= new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
+                return _parameters;
+            }
+            set => _parameters = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the component type. This value is required.
+        /// </summary>
+        [HtmlAttributeName(ComponentTypeName)]
+        public Type ComponentType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="Rendering.RenderMode"/>
+        /// </summary>
+        [HtmlAttributeName(RenderModeName)]
+        public RenderMode RenderMode { get; set; }
+
+        /// <inheritdoc />
+        public async override Task ProcessAsync(TagHelperContext context, TagHelperOutput output)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (output == null)
+            {
+                throw new ArgumentNullException(nameof(output));
+            }
+
+            var componentRenderer = ViewContext.HttpContext.RequestServices.GetRequiredService<IComponentRenderer>();
```

Comment:
Should this be a constructor parameter?

Reply:
I didn't want to make the utility type public as yet, at least until we got enough feedback about how users want to use it and how it fits for the client scenarios that were punted until 5.1. This forces the interface to not be a ctor parameter.